Improved the .editorconfig as discussed with the contributors.

Lucasharskamp opened the pull request:
As discussed in https://github.com/MonoGame/MonoGame/issues/8899, here's a simple .editorconfig to start with.

1) XML Summary mistakes, except for ''XML summaries are missing entirely'' are now marked as errors, not warnings; this is because it can brick exports to the documentation website, and besides, a good API contract requires proper formatting.
2) Added a few other common issues as warnings.

This has been tested against all solutions to check for errors that now spring up; a few malformed XML summaries were detected and dealt with:
1) Color.ToHS didn't have documentation for all its parameters
2) TextInputEventArgs' XML summary fails in Android builds because the element it's pointing at does not exist in that configuration (due to macro #if and #endif statements)
3) ImmutableVertexInputLayout refers to something called InputLayoutCache which no longer exists. Well, ImmutableVertexInputLayout itself is internal and no longer in use, so it might be an idea to remove it entirely. But that's not in the scope of this task nor do I know if it may be used elsewhere, so I left it be.
4. A number of classes in the test suite were without a namespace. However, as those are only there for performing a number of unittests, they're #pragma disabled in this case.
5. A number of XML summaries in the content pipeline had incorrect references to parameters or to an enum value that has since been renamed.

**As a result of this PR, you might notice _a ton more warnings popping up. This is fine._**

Examples:
<img width="1107" height="417" alt="image" src="https://github.com/user-attachments/assets/d79103d4-592f-4bcd-96ab-560093d6b8fa" />

In this case, the string is serialized based on the culture of the local machine, which causes comma's as the delimiter on many European desktops and dots in invariant cultures or US desktops. In this case it's relatively harmless, but this is the pinnacle of "but it works on my machine!" bug. 

<img width="694" height="320" alt="image" src="https://github.com/user-attachments/assets/95287a9c-65b7-4a15-a92d-940ed864950a" />

In this case it might be less harmless, based on what the user is doing with the output. In most cases, a simple `CultureInfo.InvariantCulture` ensures the same behavior on all machines.

Lucasharskamp committed:
Improved the .editorconfig as discussed with the contributors.
Edited a few files that didn't adhere to the new rules (minor XML summary changes)

Lucasharskamp committed:
Merge branch 'develop' into minor/editorconfig_update

Lucasharskamp committed:
Fixed several broken documentation references in the pipeline tools
Fixed missing namespace for a number of classes in the unittests.

SimonDarksideJ commented:
Great work adding the config to highlight issues.  And an even bigger thank you for continuing to also resolve the issues identified @Lucasharskamp !

Lucasharskamp committed:
1) Fixed typo in editorconfig
2) Reverted requirement for namespaces for the test classes alone, since their unittests depend on them.

Lucasharskamp committed:
Fixed disable misspelling

Lucasharskamp commented on .editorconfig:
```diff
@@ -10,3 +10,35 @@ trim_trailing_whitespace = true
+#use StringBuilder.Append(char) for single character strings.
+dotnet_diagnostic.CA1834.severity = warning
+
+#specificy StringComparison for correctness and performance.
```
Edited as requested.

SimonDarksideJ commented:
Is this ready to merge now @Lucasharskamp ?  Or do you have anything pending on this PR?

Lucasharskamp commented:
@SimonDarksideJ It's ready to go.
